fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -35,37 +35,43 @@ class ExponentiatedGradient(BaseEstimator, MetaEstimatorMixin):
         violation is at most :code:`2*(eps+best_gap)`
     :type eps: float
 
-    :param T: Maximum number of iterations
-    :type T: int
+    :param max_iterations: Maximum number of iterations
+    :type max_iterations: int
 
     :param nu: Convergence threshold for the duality gap, corresponding to a
         conservative automatic setting based on the statistical uncertainty in measuring
         classification error
     :type nu: float
 
-    :param eta_mul: Initial setting of the learning rate
-    :type eta_mul: float
+    :param learning_rate: Initial setting of the learning rate
+    :type learning_rate: float
+
+    :param run_lp_step: if True each step of exponentiated gradient is followed by the saddle
+        point optimization over the convex hull of classifiers returned so far; default True
+    :type run_lp_step: bool
     """
 
-    def __init__(self, estimator, constraints, eps=0.01, T=50, nu=None, eta_mul=2.0):  # noqa: D103
-        self._estimator = estimator
-        self._constraints = constraints
-        self._eps = eps
-        self._T = T
-        self._nu = nu
-        self._eta_mul = eta_mul
-
-        self._best_gap = None
-        self._predictors = None
-        self._weights = None
-        self._last_t = None
-        self._best_t = None
-        self._n_oracle_calls = 0
-        self._n_oracle_calls_dummy_returned = 0
-        self._oracle_execution_times = None
-        self._lambda_vecs = pd.DataFrame()
-        self._lambda_vecs_LP = pd.DataFrame()
-        self._lambda_vecs_lagrangian = pd.DataFrame()
+    def __init__(self, estimator, constraints, eps=0.01, max_iterations=50, nu=None,
+                 learning_rate=2.0, run_lp_step=True):  # noqa: D103
+        self.estimator = estimator
+        self.constraints = constraints
+        self.eps = eps
+        self.max_iterations = max_iterations
+        self.nu = nu
+        self.learning_rate = learning_rate
+        self.run_lp_step = run_lp_step
+
+        self.best_gap_ = None
+        self.predictors_ = None
+        self.weights_ = None
+        self.last_t_ = None
+        self.best_t_ = None
+        self.n_oracle_calls_ = 0
+        self.n_oracle_calls_dummy_returned_ = 0
+        self.oracle_execution_times_ = None
+        self.lambda_vecs_ = pd.DataFrame()
+        self.lambda_vecs_LP_ = pd.DataFrame()
+        self.lambda_vecs_lagrangian_ = pd.DataFrame()
 
```

Comment:
Actually, I think that we will need to tweak these names:
* `lambda_vecs_` -> `lambda_vecs_EG_` [the dimensionality of EG and LP variants is `n_iter_`]
* `lambda_vecs_lagrangian_` ->`lambda_vecs_` [this is because it matches the dimension of `predictors_` and `weights_`]

Reply:
Done!